hydromt_fiat/workflows/gis.py

```diff
@@ -304,23 +304,31 @@ def do_geocode(geolocator, xycoords, attempt=1, max_attempts=5):
         raise
 
     
-def locate_from_bounding_box(bounding_box):
+def locate_from_exposure(buildings):
     geolocator = Nominatim(user_agent="hydromt-fiat")
-
-    search = [
-        (bounding_box[1], bounding_box[0]),
-        (bounding_box[1], bounding_box[2]),
-        (bounding_box[3], bounding_box[0]),
-        (bounding_box[3], bounding_box[2]),
-    ]
-    # Find the county and state of the corner points (corners of the bounding box) of the exposure data
+    
+    # Filter buildings to reduce size
+    if len(buildings.geometry) > 100000:
+        buildings_filtered = buildings.geometry[::10000] 
+    elif len(buildings.geometry) > 10000:
+        buildings_filtered = buildings.geometry[::1000] 
+    elif len(buildings.geometry) > 1000:
+        buildings_filtered = buildings.geometry[::100] 
```

Comment:
here I think you should check for <, since now essentially always the last one would be true, right?